azure-pipelines.yml

```diff
@@ -100,64 +130,86 @@ stages:
               location: ${{ parameters.location }}
               stack: 'genesis'
               project: $(project)
-              tfversion: $(tfversion)
+              serviceConnection: $(serviceConnection)
               tfInitSub: ${{ variables.tfInitSub }}
               builtFrom: $(Build.Repository.Name)
               product: ${{ variables.product }}
+              action: ${{ parameters.action }}
 
   - stage: Network
     displayName: 'Network'
-    condition: ne('${{ parameters.Action }}', 'Destroy')
+    condition: ne('${{ parameters.action }}', 'destroy')
     jobs:
       - job: NetworkRg
         steps:
-          - template: pipeline-steps/deploy-service.yaml
+          - template: steps/set-build-repo-suffix-env-var.yaml@cnp-azuredevops-libraries
+          - template: steps/terraform.yaml@cnp-azuredevops-libraries
             parameters:
+              overrideAction: ${{ parameters.action }}
               environment: ${{ parameters.env }}
+              component: 'network-rg'
               location: ${{ parameters.location }}
-              stack: 'network-rg'
-              project: $(project)
-              tfversion: $(tfversion)
-              tfInitSub: ${{ variables.tfInitSub }}
-              builtFrom: $(Build.Repository.Name)
+              serviceConnection: $(serviceConnection)
```

Comment:
doesn't this stage need to use the `OPS-APPROVAL-GATE-MGMT-ENVS-CNP` service connection? (to do network peering against any subscription)

Reply:
Correct. Changed the network stage to use the right service connection i.e. `OPS-APPROVAL-GATE-MGMT-ENVS-CNP`